### PR TITLE
Fix email not working issue

### LINKED ISSUE #280: Email not working
Using Power Automate to generate a secret, everything says successful and I see the secrets created on my account, however the email isn't sent.  I have tried multiple emails and never received any.

### PR DESCRIPTION
**Pull request by delano**
## **User description**
Cherry picked from #307

This pull request fixes the issue with the email functionality not working properly. It includes improvements to email link accessibility, log handling, reset password request, email from address, and documentation for locale handling in the API base. These changes should ensure that emails are sent successfully and improve the overall email functionality.

Fixes #280, #285

___

## **Type**
Bug fix, Documentation

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix
documentation
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>base.rb</strong><dd><code>Improve locale handling and logging in API base</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

lib/onetime/app/api/base.rb
<li>Improved locale handling in the API base<br> <li> Added comments to explain the locale checking logic<br> <li> Removed verbose debug logging<br>

</details>

  </td>
  <td><a href="https://github.com/onetimesecret/onetimesecret/pull/308/files#diff-d373c9db90dbcb4a66720f177566ef4757dcce10a3b3e79ba7df31e1f82b86c7">+6/-3</a>&nbsp; &nbsp; &nbsp; </td>
</tr>
</table></td></tr><tr><td><strong>Bug fix
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>helpers.rb</strong><dd><code>Improve log handling for rate limiting</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

lib/onetime/app/helpers.rb
<li>Obscured customer identifiers in rate limiting logs<br> <li> Replaced verbose debug logging with targeted error handling<br>

</details>

  </td>
  <td><a href="https://github.com/onetimesecret/onetimesecret/pull/308/files#diff-e94d14dd7ae26375167811c7ec80077c552df686cbd9b62877e36f299e28725a">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>
</table></td></tr>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by onetimesecret** [comment]
**[PR Description](https://github.com/onetimesecret/onetimesecret/pull/308)** updated to latest commit (https://github.com/onetimesecret/onetimesecret/commit/82072fa4f14f43294060a178886aa4cd93b2478b)

**Comment by onetimesecret** [comment]
## PR Review

<table>
<tr>
<tr><td> ⏱️&nbsp;<strong>Estimated&nbsp;effort&nbsp;to&nbsp;review [1-5]</strong></td><td>

3, because the PR includes changes to multiple files and components, including the API base, helpers, and email functionality. The changes seem to address some bugs and improve logging and security, which require a moderate amount of review effort.


</td></tr>
<tr><td> 🧪&nbsp;<strong>Relevant tests</strong></td><td>

No


</td></tr>
<tr><td> 🔍&nbsp;<strong>Possible issues</strong></td><td>

No

</td></tr>
<tr><td> 🔒&nbsp;<strong>Security concerns</strong></td><td>

No

</td></tr>
</table>

File: lib/onetime/app/api/base.rb [hunk 1]
```diff
@@ -49,10 +49,14 @@ def authorized allow_anonymous=false
...
 
-      # Find the locale of the request based on req.env['rack.locale']
+      # Find the locale of the request based on req.env['rack.locale'],
       # which is set automatically by Otto v0.4.0 and greater.
-      # If `locale` is specifies it will override if available.
+      #
+      # If `locale` is specified, it will override if available.
       # If the `local` query param is set, it will override.
+      #
+      # @param locale [String] the locale to use, defaults to nil
```
Comment (onetimesecret) [inline lib/onetime/app/api/base.rb:58]: Consider adding type hints to the `check_locale!` method parameters and return value to improve code readability and maintainability. [medium]

Comment (delano) [reply lib/onetime/app/api/base.rb:58]: Created a ticket #309. We'll want to review type hints comprehensively for Ruby 3+. 🤩

**Pull request merged**